### PR TITLE
Upgrade ag-Grid v23 + use native ag flex columns

### PR DESCRIPTION
**Pull request by peterjcoffman**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- N/A Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ to review any possible breaking changes to direct/custom use of ag-Grid APIs and
 ### 🎁 New Features
 
 * `AppOption` configs now accept an `omit` property for conditionally excluding options.
+* Grid column `flex` prop will now accept numbers, with available space divided between flex columns
```
Comment (lbwexler) [inline CHANGELOG.md]: not to be a stickler but this is a 'property', or 'config' or 'param' anything really other than 'prop'  :-)

**Comment by lbwexler** [comment]
This one is exciting.  Thing we need a couple of usages in toolbox to try it (whether public or private)

File: CHANGELOG.md (hunk 1 above)
Comment (peterjcoffman) [reply CHANGELOG.md]: Right you are.

**Comment by lbwexler** [comment]
I feel pretty strongly that this should WAIT until Hoist 30.  

We need to stabilize the agGrid upgrade that we have already pushed in to v29 fairly aggressively and live with that for one release.

**Comment by amcclain** [comment]
Should we close this PR for now, leaving the branch open, given that we need to wait until we are confident enough to bring back the ag-Grid upgrade?

Alternatively we could apply our WIP tag - either way.  

Mainly just confirming that we _cannot_ merge this into develop at this time because develop went back to ag-Grid v21.  (Correct me of course if I am wrong here.)

**Comment by lbwexler** [comment]
Good point -- marked with WIP.  Also renamed to reflect its true nature.

**Comment by amcclain** [comment]
Great thanks - although this doesn't actually have the rest of the upgrade, so it's a bit of an odd duck in that way. 😄

**Comment by lbwexler** [comment]
@peterjcoffman -- we discussed porting the full ag-grid upgrade to this branch.  Can we ensure that is the case?  I renamed it now thinking we had done that.

**Comment by peterjcoffman** [comment]
> @peterjcoffman -- we discussed porting the full ag-grid upgrade to this branch. Can we ensure that is the case? I renamed it now thinking we had done that.

This did not have the full ag-Grid upgrade when you renamed it, but now it does.

**Comment by lbwexler** [comment]
Update:  Ag-Grid now has a reproducible error case.

Gil mentioned that he thought they had a fix in next version and will be getting back to us.

**Comment by lbwexler** [comment]
Note from Gil -- the fix we are waiting from from ag-Grid for this version is on its way

**Comment by lbwexler** [comment]
According to the changelog on ag site AG-3852 is now fixed.

**Comment by amcclain** [comment]
As mentioned over Slack a little while ago, I am reviewing this in detail now.  Will push some minor tweaks but am eager to merge and move forwards with this new update.

Will post any additional findings or questions shortly....

**Comment by amcclain** [comment]
Hi - I banged on this for a little while and it was looking good to me.  Please review minor updates in 30ce55bc3e24386708f427e783a15489e5912f4b - let me know if those look reasonable.

I did not test on mobile yet and should jump to another set of tasks at this point in the morning. @TomTirapani did you have a chance to do some real on-device testing yet?  

So great to remove all those resize calls.  I hope that we don't see any unexpected performance issues w/whatever the grid is doing internally now to support flexing, but I am hopeful that if anything this will be a performance pickup.

I tweaked the Toolbox sampleGrid locally to specify two of the cols with different flexes and set min/max widths. Worked great. Did not seem worth a PR there but would make the same change once we merge.

**Comment by TomTirapani** [comment]
Played around with this in Toolbox on device - didn't find any issues in Chrome, Safari or client secure browsers (all iOS).

**Comment by lbwexler** [comment]
Anselm -- glad you actually used the flex in toolbox.  Was going to ask about that  -- I know peter had showed me some examples there, but somehow the never got put into a PR.

Tom -- I found a regression in tiny mode -- looks like the tree affordances no longer fit within the row.  Can you take a look?

**Comment by amcclain** [comment]
If I hit up http://localhost:3000/admin/tests/cube I see an error in my console:

Screen Shot 2020-03-25 at 11 47 57 AM

Believe similar issues referenced above - sorry to drop this in here and run but am juggling a few things, didn't want to ignore/forget however....

**Comment by lbwexler** [comment]
I am also seeing some problematic behavior with the flexing of a single name column. See the main portfolio grid in the portfolio example. 

+ We seem to get an initial flash with the new ag-grid implementation
+Behavior has definitely changed when we do the resize operation in the context menu.  No longer apportioning all of the empty space to the name column, but seems to be making it the minSize and apportioning the extra space to the right of all columns

**Comment by TomTirapani** [comment]
Pushed a fix for the tree grid arrow alignment.

It turns out our tree grid styles weren't being applied since this change: https://github.com/xh/hoist-react/commit/c090811f221b27d348f6c847a1a4ea1b88132cc2 (not directly related, but also fixed).

I also tweaked the arrow spacing for the smaller sizing modes, since they seemed to large for the smaller sizing modes.

Looking at the other issues now.

**Comment by TomTirapani** [comment]
I've been trying to reproduce the issue on the cube test page, but can't seem to reproduce it...

**Comment by TomTirapani** [comment]
Regarding the autosize + flex issue: https://www.ag-grid.com/javascript-grid-resizing/#column-flex
```
If you manually resize a column with flex either via API or dragging the resize handle,
flex will be automatically disabled for that column.
```
We are already setting `resizable: false` on flex columns, and have removed calls to `sizeColumnsToFit();`, but calling `autoSizeColumns()` on flex columns would disable flex.

I've pushed a change to skip any flex columns before calling `autoSizeColumns()`

**Comment by lbwexler** [comment]
Tom -- thanks for the great fix to flex columns.

I also cannot reproduce the console errors Anselm posted

Tree Styles look mostly great -- a lot less wasted space, with everything shifted over substantially to the left.  But it seems like we may need to change two other related things:

+ The  text for a leaf node has *not* shifted over to the left.  So it now seems weirdly too far to the right relative to it  
+ The text for the Total header has *not* shifted over to the left.  So it now doesn't align with the data.

**Comment by amcclain** [comment]
We spent a lot of time on the tree grid styling and indentation last month - it looks like maybe something has regressed.  I adjusted the `--xh-grid-tree-indent-px` we added (which was working well) and it no longer appears to have an effect. 

Possible selector specificity issue with these classes no longer being applied as desired? 

https://github.com/xh/hoist-react/blob/develop/cmp/grid/Grid.scss#L95

**Comment by amcclain** [comment]
I don't see the console errors now myself either.

I don't know what to tell you guys - you see the screenshot.  I had a clean sandbox.  Something else must have changed. I didn't imagine it.

**Comment by TomTirapani** [comment]
Fixed the indentation styling, across total row, tree groups and 'mixed' trees:

Screenshot 2020-03-26 at 12 37 00

**Pull request merged**